### PR TITLE
Bz1933918

### PR DESCRIPTION
**Pull request by stevsmit**
Adds a module for passthrough routes to the Routes assembly. 

For 4.5+

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1933918

Preview at https://deploy-preview-30957--osdocs.netlify.app/openshift-enterprise/latest/networking/routes/secured-routes.html#nw-ingress-creating-a-passthrough-route_secured-routes

@zhaozhanqi Please review. Thank you.

### CODE REVIEW

**Comment by lihongan** [comment]
Looks great! Thank you @stevsmit

**Comment by bmcelvee** [comment]
@wgordon17 would you mind giving this PR a quick review, please? Thank you!

File: modules/nw-ingress-creating-a-passthrough-route.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  to:
+    kind: Service
+    name: frontend
+----
+<1> The name of the object, which is limited to 63 characters.
+<2> The `*termination*` field is set to `passthrough`. No other encryption fields are needed.
++
+The destination pod is responsible for serving certificates for the
+traffic at the endpoint. This is currently the only method that can support
+requiring client certificates (also known as two-way authentication).
```
Comment (bmcelvee) [inline modules/nw-ingress-creating-a-passthrough-route.adoc:47]: Sorry I didn't catch this before! The IBM style guide advises against using parenthesis.  

Suggested edits: 

This is currently the only method that can support
requiring client certificates, also known as two-way authentication.

The only valid values are `None`, empty for disabled, or `Redirect`.

**Comment by bmcelvee** [comment]
LGTM!

File: modules/nw-ingress-creating-a-passthrough-route.adoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
+// Module included in the following assemblies:
+//
+// * ingress/routes.adoc
+
+[id="nw-ingress-creating-a-passthrough-route_{context}"]
+= Creating a passthrough route
+
+You can configure a secure route using passthrough termination by using the `oc create route` command. With passthrough termination, encrypted traffic is sent straight to the destination without the router providing TLS termination. Therefore no key or certificate is required.
```
Comment (wgordon17) [inline modules/nw-ingress-creating-a-passthrough-route.adoc:8]:
```suggestion
You can configure a secure route using passthrough termination by using the `oc create route` command. With passthrough termination, encrypted traffic is sent straight to the destination without the router providing TLS termination. Therefore no key or certificate is required on the route.
```

The key and certificate are still required on the pod

File: modules/nw-ingress-creating-a-passthrough-route.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+apiVersion: v1
+kind: Route
+metadata:
+  name: route-passthrough-secured <1>
+spec:
+  host: www.example.com
+  port:
+    targetPort: 8080
+  tls:
+    termination: passthrough <2>
```
Comment (wgordon17) [inline modules/nw-ingress-creating-a-passthrough-route.adoc:37]:
```suggestion
    termination: passthrough <2>
    insecureEdgeTerminationPolicy: None <3>
```

File: modules/nw-ingress-creating-a-passthrough-route.adoc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  to:
+    kind: Service
+    name: frontend
+----
+<1> The name of the object, which is limited to 63 characters.
+<2> The `*termination*` field is set to `passthrough`. No other encryption fields are needed.
```
Comment (wgordon17) [inline modules/nw-ingress-creating-a-passthrough-route.adoc:43]:
```suggestion
<2> The `*termination*` field is set to `passthrough`. This is the only required `tls` field.
<3> Optional `insecureEdgeTerminationPolicy`. The only valid values are `None`, `Redirect`, or empty for disabled.
```

File: modules/nw-ingress-creating-a-passthrough-route.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
++
+The destination pod is responsible for serving certificates for the
+traffic at the endpoint. This is currently the only method that can support
+requiring client certificates, also known as two-way authentication.
++
+[NOTE]
```
Comment (wgordon17) [inline modules/nw-ingress-creating-a-passthrough-route.adoc:49]: This note is no longer necessary

File: networking/routes/secured-routes.adoc [hunk 6]
```diff
@@ -5,8 +5,7 @@ include::modules/common-attributes.adoc[]
 
 toc::[]
 
-The following sections describe how to create re-encrypt and edge routes with
-custom certificates.
+Secure routes provide the ability to use several types of TLS termination to serve certificates to the client. The follow sections describe how to create re-encrypt, edge, and passthrough routes with custom certificates.
```
Comment (wgordon17) [inline networking/routes/secured-routes.adoc:8]:
```suggestion
Secure routes provide the ability to use several types of TLS termination to serve certificates to the client. The following sections describe how to create re-encrypt, edge, and passthrough routes with custom certificates.
```

**Review by wgordon17 — CHANGES_REQUESTED** [review]

**Comment by wgordon17** [comment]
LGTM, thanks!

**Review by wgordon17 — APPROVED** [review]

**Comment by stevsmit** [comment]
@wgordon17 @bmcelvee All edits have been made! Thank you both.

**Pull request merged**